compiler/rustc_const_eval/src/const_eval/machine.rs

```diff
@@ -65,6 +65,9 @@ pub struct CompileTimeMachine<'tcx> {
     /// storing the result in the given `AllocId`.
     /// Used to prevent reads from a static's base allocation, as that may allow for self-initialization loops.
     pub(crate) static_root_ids: Option<(AllocId, LocalDefId)>,
+
+    /// A cache of "data range" computations for unions (i.e., the offsets of non-padding bytes).
```

Comment:
This seems like the sort of thing that would normally be a query. Is there a reason it isn't?

Reply:
I'm not sure we even want this cached during a normal compilation (but it probably won't hurt much either), and the `RangeSet` type is currently only declared in the interpreter so can't be used as a query. Also I am largely unfamiliar with the query system so adding a query is scary. ;)

Reply:
If we have many constants of the same (union) type, then we would encounter it repeatedly and caching might be worth it. But we don't do validation during const-eval by default so the cache is not nearly as necessary as for Miri.

I can see how making this a query makes sense. But it would probably be a query we wouldn't want to serialize for incremental (I assume there's a way to do that), and as I said I am not sure where we would then put the `RangeSet`. Probably `rustc_data_structures`, but that seems odd for such a specific data structure. I'm open for suggestions though. :)

Reply:
You've made a good case that we shouldn't make this into a query in this PR, I'm filing away this conversation for the next time I'm squinting at interpreter profiles :)